modules/standard/Filerator.chpl

```diff
@@ -192,43 +200,114 @@ iter wordexp(pattern="*") {
    By default, it will list all files/directories in the current directory
 */
 
-iter glob(pattern="*") {
+module Chpl_rt_glob {
   extern type glob_t;
   extern proc chpl_glob(pattern:c_string, flags: c_int, 
                         ref ret_glob:glob_t):c_int;
   extern proc chpl_glob_num(x:glob_t): size_t;
   extern proc chpl_glob_index(x:glob_t, idx:size_t): c_string;
   extern proc globfree(ref glb:glob_t);
+}
 
-  var glb : glob_t;
+iter glob(pattern="*") {
+  var glb : Chpl_rt_glob.glob_t;
 
-  const err = chpl_glob(pattern:c_string, 0, glb);
-  const num = chpl_glob_num(glb);
+  const err = Chpl_rt_glob.chpl_glob(pattern:c_string, 0, glb);
+  const num = Chpl_rt_glob.chpl_glob_num(glb);
   if (num) then
```

Comment:
I think you can safely get rid of this check and the loop after it simply won't execute. Doing so will allow this iterator to be inlined for the zippered case as well. Similarly removing the check for the follower will allow it to be inlined in the zippered case (I checked the generated code for these two cases.)

I'm not sure if you want to keep it in the standalone case... The standalone will be directly inlined (and clearly will never be zippered) so it should be fine to leave it, but I *think* it could be removed. Similar story for the leader.